lib/manageiq/api/common/filter.rb

```diff
@@ -23,30 +23,24 @@ def apply
               type = determine_string_attribute_type(attribute)
               send(type, k, v)
             else
-              errors.add("unsupported attribute type for: #{k}")
+              errors << "unsupported attribute type for: #{k}"
```

Comment:
@bdunne This pattern seems to be like how ActiveRecord does the validations of all parameters and collects them and raises a single error if the @errors array is not empty. With the other exception handling we are raising a chain of exceptions and traversing the exception cause to build an array of error messages.
https://github.com/rails/rails/blob/6d68bb5f695414b1801c1c3952ef955a5b0b6c6a/activerecord/lib/active_record/validations.rb#L21